tests/scene/test_split_container.h

```diff
@@ -0,0 +1,1409 @@
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_SPLIT_CONTAINER_H
+#define TEST_SPLIT_CONTAINER_H
+
+#include "scene/gui/split_container.h"
+#include "scene/main/window.h"
+
+#include "tests/test_macros.h"
+
+namespace TestSplitContainer {
+
+static inline void check_positions(SplitContainer *sc, Vector<int> positions, int sep, bool horizontal = true) {
+	// Checks the rects of each split container child.
+	CHECK(sc->get_child_count(false) == positions.size() + 1);
+	int last_pos = 0;
+	for (int i = 0; i < sc->get_child_count(false); i++) {
+		// Assuming there is no invalid children.
+		Control *c = Object::cast_to<Control>(sc->get_child(i, false));
+		int pos = i >= positions.size() ? sc->get_size()[horizontal ? 0 : 1] : positions[i];
+		Rect2 rect;
+		if (horizontal) {
+			rect.size.y = sc->get_size().y;
+			rect.position.x = last_pos;
+			rect.size.x = pos - last_pos;
+		} else {
+			rect.size.x = sc->get_size().x;
+			rect.position.y = last_pos;
+			rect.size.y = pos - last_pos;
+		}
+		CHECK_MESSAGE(c->get_rect() == rect, vformat("Child %s is the wrong size.", i));
+		last_pos = pos + sep;
+	}
+}
+
+static inline void check_position(SplitContainer *sc, int position, int sep, bool horizontal = true) {
+	check_positions(sc, { position }, sep, horizontal);
+}
+
+static inline void check_positions_rtl(SplitContainer *sc, Vector<int> positions, int sep) {
+	// Checks the rects of each split container child. Right to left layout.
+	CHECK(sc->get_child_count(false) == positions.size() + 1);
+	int last_pos = sc->get_size().x;
+	for (int i = 0; i < sc->get_child_count(false); i++) {
+		// Assuming there is no invalid children.
+		Control *c = Object::cast_to<Control>(sc->get_child(i, false));
+		int pos = i >= positions.size() ? 0 : sc->get_size().x - positions[i];
+		Rect2 rect;
+		rect.size.y = sc->get_size().y;
+		rect.position.x = pos;
+		rect.size.x = last_pos - pos;
+		CHECK_MESSAGE(c->get_rect() == rect, vformat("Child %s is the wrong size.", i));
+		last_pos = pos - sep;
+	}
+}
+
+static inline void check_position_rtl(SplitContainer *sc, int position, int sep) {
+	check_positions_rtl(sc, { position }, sep);
+}
+
+TEST_CASE("[SceneTree][SplitContainer] add children") {
```

Comment:
I didn't know that test names should be capitalized, since TextEdit tests don't. I didn't do it for TabBar or TabContainer tests either, but I guess those can be fixed another time.

Reply:
Generally they should, older tests are out of date, so best to fix for the new ones and keep the rest tracked for future fixes!